mmv1/third_party/terraform/resources/resource_dataproc_cluster.go.erb

```diff
@@ -638,6 +645,44 @@ func resourceDataprocCluster() *schema.Resource {
 											},
 										},
 									},
+
+									"reservation_affinity": {
+										Type:         schema.TypeList,
+										Optional:     true,
+										AtLeastOneOf: gceClusterConfigKeys,
+										Computed:     true,
+										MaxItems:     1,
+										Description:  `Reservation Affinity for consuming Zonal reservation.`,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"consume_reservation_type": {
+													Type:         schema.TypeString,
+													Optional:     true,
+													AtLeastOneOf: reservationAffinityKeys,
+													ForceNew:     true,
+													ValidateFunc: validation.StringInSlice([]string{"TYPE_UNSPECIFIED", "NO_RESERVATION", "ANY_RESERVATION", "SPECIFIC_RESERVATION"}, false),
+													Default:      false,
```

Comment:
Does this behave as expected when a value is not provided? It looks like the wrong type for this field since it is a string. I would consider not including a default here unless it is needed by the API.